src/internal/connector/graph/consts.go

```diff
@@ -25,3 +32,87 @@ const (
 	// given endpoint.
 	PreviousPathFileName = "previouspath"
 )
+
+// ---------------------------------------------------------------------------
+// Runtime Configuration
+// ---------------------------------------------------------------------------
+
+type parallelism struct {
+	collectionPool int
+	itemFetch      int
+	prefetchPool   int
+}
+
+func (p parallelism) CollectionPoolSize() int {
+	return p.collectionPool
+}
+
+func (p parallelism) CollectionPoolOverride(ctx context.Context, override int) int {
+	logger.Ctx(ctx).Infow(
+		"collection pool parallelism",
+		"default_parallelism", p.itemFetch,
+		"requested_paralellism", override)
+
+	if override < 1 || (p.collectionPool > 0 && override > p.collectionPool) {
```

Comment:
I know it's a simple comparison, but I kind of feel like we should have a separate helper function for this. It would make implementing minor changes to stuff a bit easier in the long run